### PR TITLE
dev-embedded/ponyprog-3.0.0: new package

### PR DESCRIPTION
**Pull request by ktrace**
new version ponyprog, with necessary changes in app-editors/qhexedit2

### CODE REVIEW

File: dev-embedded/ponyprog/ponyprog-3.0.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	dev-qt/qtgui:5
+	dev-qt/qtmultimedia:5
+	dev-qt/qtprintsupport:5
+	app-editors/qhexedit2"
+
+DEPEND="${RDEPEND}"
+
+src_prepare() {
+	cmake-utils_src_prepare
+}
```
Comment (SoapGentoo) [inline dev-embedded/ponyprog/ponyprog-3.0.0.ebuild]: not needed

File: dev-embedded/ponyprog/ponyprog-3.0.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+src_compile() {
+	cmake-utils_src_compile
+}
```
Comment (SoapGentoo) [inline dev-embedded/ponyprog/ponyprog-3.0.0.ebuild]: not needed

File: dev-embedded/ponyprog/ponyprog-3.0.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
```
Comment (SoapGentoo) [inline dev-embedded/ponyprog/ponyprog-3.0.0.ebuild]: quotes not necessary

File: dev-embedded/ponyprog/ponyprog-3.0.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+DESCRIPTION="Serial device programmer"
+HOMEPAGE="https://github.com/lancos/ponyprog/"
+SRC_URI="https://github.com/lancos/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="LGPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+PATCHES=( "${FILESDIR}/${P}-qhexedit-dynamic-link.patch" )
```
Comment (SoapGentoo) [inline dev-embedded/ponyprog/ponyprog-3.0.0.ebuild]: after `DEPEND` pls

File: dev-embedded/ponyprog/ponyprog-3.0.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+LICENSE="LGPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND="dev-qt/qtcore:5
+	dev-qt/qtgui:5
+	dev-qt/qtmultimedia:5
+	dev-qt/qtprintsupport:5
+	app-editors/qhexedit2:0="
```
Comment (titanofold) [inline dev-embedded/ponyprog/ponyprog-3.0.0.ebuild]: Alphabetical order, please.

For future work you may do: It's easier to read when dependencies are grouped by unversioned, versioned, and use-conditional.

File: app-editors/qhexedit2/Manifest [hunk 6]
```diff
@@ -1 +1 @@
-DIST qhexedit2-0.8.4.tar.gz 340766 BLAKE2B af24031fdd9f8d677f7e8ab8fd69667852dd971db30f5406b9aaaa08bf7f2dc692c87e99c141f355adc82c8226b5b29f53ce07ba63656d866f8688cb2cbc67f2 SHA512 8ed831d75c8ec1897dfd7ef497423908b26b0f56e6577928ed035654e52b437f64edd61cfcc1de6eaf5d1f29f9dc97da84d2edd7f427e3c09c9d75d7bc6504ff
+DIST qhexedit2-0.8.4.tar.gz 341967 BLAKE2B 61ba4bd760ca0ea79aa10eece6bda5a9847e15d44e7edb329a9bedd6bcaad0a37ac7933b9238b0e0228e01258fc8a802c99a251141a6e29642fe8e5fc175889e SHA512 e42bfe1adb5f17d8285f3f9966f49550d360c4a98fc7f6bcbb29d66628ee5c781c5ad4170115ebf86835a296932ed331c583fdbcd91e944412ad3c173458e13a
```
Comment (titanofold) [inline app-editors/qhexedit2/Manifest]: Whoa! Why'd this change?

File: app-editors/qhexedit2/qhexedit2-0.8.4.ebuild [hunk 7]
```diff
@@ -11,7 +11,7 @@ inherit distutils-r1 qmake-utils
 
 DESCRIPTION="Hex editor library, Qt application written in C++ with Python bindings"
 HOMEPAGE="https://github.com/lancos/qhexedit2/"
-SRC_URI="https://github.com/lancos/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+SRC_URI="https://github.com/lancos/${PN}/archive/ponyprog.tar.gz -> ${P}.tar.gz"
```
Comment (titanofold) [inline app-editors/qhexedit2/qhexedit2-0.8.4.ebuild]: Are we only masquerading as qhexedit2 now?

File: app-editors/qhexedit2/Manifest (hunk 6 above)
Comment (ktrace) [reply app-editors/qhexedit2/Manifest]: 'cause right branch selected now.

File: app-editors/qhexedit2/qhexedit2-0.8.4.ebuild (hunk 7 above)
Comment (ktrace) [reply app-editors/qhexedit2/qhexedit2-0.8.4.ebuild]: qhexedit2 has branch ponyprog. in first ebuild version i was select incorrect branch, ponyprog 3.0.0 can't build with it.

File: dev-embedded/ponyprog/ponyprog-3.0.0.ebuild (hunk 5 above)
Comment (ktrace) [reply dev-embedded/ponyprog/ponyprog-3.0.0.ebuild]: done

File: app-editors/qhexedit2/qhexedit2-0.8.4.ebuild (hunk 7 above)
Comment (Civil) [reply app-editors/qhexedit2/qhexedit2-0.8.4.ebuild]: In fact this becomse -9999.ebuild, beacuse you are pointing it to the HEAD of ponyprog branch.
Also please note that master and ponyprog branches are even in terms of commits.

Comment (Civil) [reply app-editors/qhexedit2/qhexedit2-0.8.4.ebuild]: And by -9999 I mean that it will also break next time somebody will commit to that branch with "hash mismatch".

File: app-editors/qhexedit2/qhexedit2-0.8.4.ebuild [hunk 8]
```diff
@@ -20,8 +20,8 @@ IUSE="doc gui python"
 REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
 
 PATCHES=(
-	"${FILESDIR}/${P}-fixtest.patch"
-	"${FILESDIR}/${P}-setup.py.patch"
+	"${FILESDIR}/${PN}-fixtest.patch"
```
Comment (SoapGentoo) [inline app-editors/qhexedit2/qhexedit2-0.8.4.ebuild]: why did you remove the version string? In general `${P}-blabla.patch` is preferred

File: app-editors/qhexedit2/qhexedit2-0.8.4_p20170719.ebuild [hunk 9]
```diff
@@ -0,0 +1,86 @@
+# Copyright 2017-2018 Gentoo Foundation
```
Comment (SoapGentoo) [inline app-editors/qhexedit2/qhexedit2-0.8.4_p20170719.ebuild]: shouldnt copyright start at 1999?

File: app-editors/qhexedit2/qhexedit2-0.8.4_p20170719.ebuild [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+)
+
+RDEPEND="
+	dev-qt/qtcore:5
+	dev-qt/qtgui:5
+	dev-qt/qtwidgets:5
+	python? ( dev-python/PyQt5[gui,widgets,${PYTHON_USEDEP}]
+			dev-python/sip[${PYTHON_USEDEP}]
+			${PYTHON_DEPS} )
+"
```
Comment (SoapGentoo) [inline app-editors/qhexedit2/qhexedit2-0.8.4_p20170719.ebuild]:
```
python? (
	dev-python/PyQt5[gui,widgets,${PYTHON_USEDEP}]
	dev-python/sip[${PYTHON_USEDEP}]
	${PYTHON_DEPS}
)
```

File: app-editors/qhexedit2/qhexedit2-0.8.4.ebuild (hunk 8 above)
Comment (ktrace) [reply app-editors/qhexedit2/qhexedit2-0.8.4.ebuild]: cause patches used both qhexedit2-0.8.4.ebuild and qhexedit2-0.8.4_p20170719.ebuild

Comment (SoapGentoo) [reply app-editors/qhexedit2/qhexedit2-0.8.4.ebuild]: then make it `qhexedit2-0.8.4-fixtest.patch` and do `"${FILESDIR}/${PN}-0.8.4-fixtest.patch"` in the qhexedit2-0.8.4_p20170719.ebuild

File: app-editors/qhexedit2/qhexedit2-0.8.4.ebuild (hunk 7 above)
Comment (SoapGentoo) [reply app-editors/qhexedit2/qhexedit2-0.8.4.ebuild]: this still has not been adressed

File: app-editors/qhexedit2/qhexedit2-0.8.4_p20170719.ebuild (hunk 9 above)
Comment (ktrace) [reply app-editors/qhexedit2/qhexedit2-0.8.4_p20170719.ebuild]: fixed

File: app-editors/qhexedit2/qhexedit2-0.8.4_p20170719.ebuild (hunk 10 above)
Comment (ktrace) [reply app-editors/qhexedit2/qhexedit2-0.8.4_p20170719.ebuild]: fixed

File: app-editors/qhexedit2/qhexedit2-0.8.4_p20170719.ebuild [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+	dev-qt/qtwidgets:5
+	python? (
+		dev-python/PyQt5[gui,widgets,${PYTHON_USEDEP}]
+		dev-python/sip[${PYTHON_USEDEP}]
+		${PYTHON_DEPS}
+		)
+"
+DEPEND="${RDEPEND}"
+
+S="${WORKDIR}/${PN}-5f3ca79dbe8c765b685636a1751d242a6bfabb7d"
```
Comment (SoapGentoo) [inline app-editors/qhexedit2/qhexedit2-0.8.4_p20170719.ebuild]: instead of this, consider using `vcs-snapshot.eclass`

File: app-editors/qhexedit2/qhexedit2-0.8.4_p20170719.ebuild [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+	fi
+}
+
+src_compile() {
+	default
+	use python && distutils-r1_src_compile
+	if use gui; then
+		cd example || die "can't cd example"
+		emake
+	fi
```
Comment (SoapGentoo) [inline app-editors/qhexedit2/qhexedit2-0.8.4_p20170719.ebuild]: change the if block to `use gui && emake -C example`

File: dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch [hunk 13]
```diff
@@ -0,0 +1,36 @@
+--- ponyprog-3.0.0/distribution/posix/CMakeLists.txt-	2018-01-18 21:12:21.504884644 +0300
++++ ponyprog-3.0.0/distribution/posix/CMakeLists.txt	2018-01-18 22:02:34.575994692 +0300
+@@ -33,25 +33,25 @@
+         ) 
+ 
+ INSTALL(DIRECTORY "${PROJECT_SOURCE_DIR}/distribution/innosetup/" 
+-        DESTINATION "share/doc/ponyprog/" 
++        DESTINATION "share/doc/ponyprog-3.0.0/"
```
Comment (SoapGentoo) [inline dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch]: As @dracwyrm suggested, please use the `GNUInstallDirs` module, use `${CMAKE_INSTALL_DOCDIR}` here and upstream the resulting patch

File: dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+-        DESTINATION "share/doc/ponyprog/" 
++        DESTINATION "share/doc/ponyprog-3.0.0/" 
+         COMPONENT "help files" 
+         DIRECTORY_PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ GROUP_EXECUTE GROUP_READ WORLD_EXECUTE WORLD_READ 
+         FILES_MATCHING PATTERN "*.jpg" PATTERN "*.html"
+         ) 
+ 
+ INSTALL(FILES "${PROJECT_SOURCE_DIR}/copyright" 
+-        DESTINATION "share/doc/ponyprog/" 
++        DESTINATION "share/doc/ponyprog-3.0.0/"
```
Comment (SoapGentoo) [inline dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch]: `${CMAKE_INSTALL_DOCDIR}`

File: dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+ # MESSAGE("PROJECT_SOURCE_DIR ${PROJECT_SOURCE_DIR}")
+ 
+-INSTALL(FILES "${PROJECT_SOURCE_DIR}/desktop/ponyprog.desktop"
+-        DESTINATION "share/applications/" 
+-        COMPONENT "application" 
+-        PERMISSIONS OWNER_WRITE OWNER_READ GROUP_READ WORLD_READ
+-        )
+-        
++#INSTALL(FILES "${PROJECT_SOURCE_DIR}/desktop/ponyprog.desktop"
++#        DESTINATION "share/applications/"
```
Comment (SoapGentoo) [inline dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch]: `${CMAKE_INSTALL_DATAROOTDIR}/applications`

File: dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+-INSTALL(FILES "${PROJECT_SOURCE_DIR}/desktop/ponyprog.desktop"
+-        DESTINATION "share/applications/" 
+-        COMPONENT "application" 
+-        PERMISSIONS OWNER_WRITE OWNER_READ GROUP_READ WORLD_READ
+-        )
+-        
++#INSTALL(FILES "${PROJECT_SOURCE_DIR}/desktop/ponyprog.desktop"
++#        DESTINATION "share/applications/" 
++#        COMPONENT "application" 
++#        PERMISSIONS OWNER_WRITE OWNER_READ GROUP_READ WORLD_READ
```
Comment (SoapGentoo) [inline dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch]: and remove this permission crap, its stupid and causes issues for us

File: dev-embedded/ponyprog/ponyprog-3.0.0.ebuild [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+	"${FILESDIR}/${P}-qhexedit-dynamic-link.patch"
+	)
+
+src_install() {
+	cmake-utils_src_install
+	keepdir /var/lock/uucp
+	fowners uucp:uucp /var/lock/uucp
+	fperms 755 /var/lock/uucp
+	doicon icons/ponyprog.png
+	make_desktop_entry ponyprog2000 PonyProg2000 ponyprog
```
Comment (SoapGentoo) [inline dev-embedded/ponyprog/ponyprog-3.0.0.ebuild]: once you fixed the CMake, you won't need this line anymore

**Comment by mgorny** [comment]
@ktrace, ping.

File: app-editors/qhexedit2/qhexedit2-0.8.4_p20170719.ebuild (hunk 11 above)
Comment (mgorny) [reply app-editors/qhexedit2/qhexedit2-0.8.4_p20170719.ebuild]: @SoapGentoo, please don't encourage use of deprecated eclasses. Since the hash needs to be known anyway, it's better to use it explicitly.

@ktrace, just move it to a common variable `EGIT_COMMIT` and reuse it both for SRC_URI and S.

**Comment by ktrace** [comment]
alive, need several days.

File: app-editors/qhexedit2/qhexedit2-0.8.4_p20170719.ebuild (hunk 11 above)
Comment (ktrace) [reply app-editors/qhexedit2/qhexedit2-0.8.4_p20170719.ebuild]: fixed

File: dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch (hunk 16 above)
Comment (ktrace) [reply dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch]: remaked. but  ${CMAKE_INSTALL_DOCDIR} has no version... must be .../usr/share/doc/ponyprog-3.0.0

File: app-editors/qhexedit2/qhexedit2-0.8.4.ebuild [hunk 18]
```diff
@@ -34,6 +34,8 @@ RDEPEND="
 "
 DEPEND="${RDEPEND}"
 
+S="${WORKDIR}/${PN}-ponyprog"
```
Comment (mgorny) [inline app-editors/qhexedit2/qhexedit2-0.8.4.ebuild]: Why are you changing the old ebuild?

File: app-editors/qhexedit2/qhexedit2-0.8.4_p20170719.ebuild (hunk 11 above)
Comment (mgorny) [reply app-editors/qhexedit2/qhexedit2-0.8.4_p20170719.ebuild]: @ktrace but please reuse `${EGIT_COMMIT}` here.

File: dev-embedded/ponyprog/Manifest [hunk 19]
```diff
@@ -1 +1,2 @@
 DIST Pony_Prog2000-2.07c.tar.gz 653300 BLAKE2B 7b20450dfefb4370fb813b5532e6ba504b6ac8e598a2e46615ba522537e91b7798f6f86fbf93320fc71be5fe756ceef3351304196da191f7e68670f75ab5f865 SHA512 33cd14fa8283ce8af8265a48df7d960c7c2779551d0f204f6728a9e1e54204f52ea608c4e6090dc56f7634a02abd9bb960230f75a3a9716b1639708a57c31809
```
Comment (mgorny) [inline dev-embedded/ponyprog/Manifest]: But it's not a new package? Just a new version?

File: dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch [hunk 20]
```diff
@@ -0,0 +1,68 @@
...
+         ) 
+ # MESSAGE("PROJECT_SOURCE_DIR ${PROJECT_SOURCE_DIR}")
+ 
+-INSTALL(FILES "${PROJECT_SOURCE_DIR}/desktop/ponyprog.desktop"
+-        DESTINATION "share/applications/" 
+-        COMPONENT "application" 
+-        PERMISSIONS OWNER_WRITE OWNER_READ GROUP_READ WORLD_READ
+-        )
+-        
++#INSTALL(FILES "${PROJECT_SOURCE_DIR}/desktop/ponyprog.desktop"
```
Comment (mgorny) [inline dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch]: Why is this commented out?

File: dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch (hunk 16 above)
Comment (mgorny) [reply dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch]: Yeah, you need to pass `-DCMAKE_INSTALL_DOCDIR` to `mycmakeargs` manually.

File: dev-embedded/ponyprog/metadata.xml [hunk 21]
```diff
@@ -5,7 +5,16 @@
 		<email>embedded@gentoo.org</email>
 		<name>Embedded Gentoo</name>
 	</maintainer>
+	<maintainer type="project">
```
Comment (mgorny) [inline dev-embedded/ponyprog/metadata.xml]: Reverse it, please. Proxy-maint always comes last.

File: dev-embedded/ponyprog/metadata.xml [hunk 22]
```diff
@@ -5,7 +5,16 @@
...
+		<email>proxy-maint@gentoo.org</email>
+		<name>Proxy Maintainers</name>
+	</maintainer>
+	<maintainer type="person">
+		<email>coyote@bks.tv</email>
+		<name>Victor Kustov</name>
+	</maintainer>
 	<upstream>
 		<remote-id type="sourceforge">ponyprog</remote-id>
+		<remote-id type="github">lancos/ponyprog</remote-id>
```
Comment (mgorny) [inline dev-embedded/ponyprog/metadata.xml]: Sort lexically (by 'type').

File: app-editors/qhexedit2/qhexedit2-0.8.4.ebuild (hunk 18 above)
Comment (mgorny) [reply app-editors/qhexedit2/qhexedit2-0.8.4.ebuild]: You still didn't answer this.

File: app-editors/qhexedit2/qhexedit2-0.8.4_p20170719.ebuild (hunk 11 above)
Comment (mgorny) [reply app-editors/qhexedit2/qhexedit2-0.8.4_p20170719.ebuild]: Also this.

File: dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch [hunk 23]
```diff
@@ -0,0 +1,70 @@
...
+ 
++INCLUDE(GNUInstallDirs)
++
+ MESSAGE("CMAKE_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX}")
+ 
+ # SET( CPACK_PACKAGE_GROUP "${PONYPROG_CPACK_PACKAGE_GROUP}" )
+ SET( CPACK_PACKAGE_LICENSE "${PONYPROG_CPACK_PACKAGE_LICENSE}" )
+ SET( CPACK_PACKAGE_SHLIBDEPS "${PONYPROG_CPACK_PACKAGE_SHLIBDEPS}" )
+ # SET( CPACK_PACKAGE_DEPENDS "${PONYPROG_CPACK_PACKAGE_DEPENDS}" )
++SET ( CMAKE_INSTALL_DOCDIR "${CMAKE_INSTALL_DATAROOTDIR}/doc/${PROJECT_NAME}-${APP_VERSION}" )
```
Comment (mgorny) [inline dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch]: Set this in ebuild, not in cmake.

File: app-editors/qhexedit2/qhexedit2-0.8.4.ebuild (hunk 18 above)
Comment (ktrace) [reply app-editors/qhexedit2/qhexedit2-0.8.4.ebuild]: was comment with rearrange and broken copyright. drop it here?

File: dev-embedded/ponyprog/Manifest (hunk 19 above)
Comment (ktrace) [reply dev-embedded/ponyprog/Manifest]: new version. fixed

File: dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch (hunk 20 above)
Comment (ktrace) [reply dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch]: cause make_session_desktop used. is it ok?

File: dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch (hunk 15 above)
Comment (ktrace) [reply dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch]: fixed

File: dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch (hunk 16 above)
Comment (ktrace) [reply dev-embedded/ponyprog/files/ponyprog-3.0.0-install_doc.patch]: added

**Comment by mgorny** [comment]
```
 * QA Notice: This package installs one or more .desktop files that do not
 * pass validation.
 * 
 * 	/usr/share/applications/ponyprog.desktop: error: (will be fatal in the future): value "ponyprog.png" for key "Icon" in group "Desktop Entry" is an icon name with an extension, but there should be no extension as described in the Icon Theme Specification if the value is not an absolute path
 * 	/usr/share/applications/ponyprog.desktop: error: value "3.0.0" for key "Version" in group "Desktop Entry" is not a known version
 * 
```

That we know about, and I can live with it being broken. However:

```
 * QA Notice: Files built without respecting CFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/bin/ponyprog
```

This is something you ought to fix.

**Comment by ktrace** [comment]
i may fix it by patch. or leave it and wait when upstream fix it. must i fix it by patch?

File: dev-embedded/ponyprog/ponyprog-3.0.0.ebuild [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+
+PATCHES=(
+	"${FILESDIR}/${P}-install_doc.patch"
+	"${FILESDIR}/${P}-qhexedit-dynamic-link.patch"
+	)
+
+src_configure() {
+	local mycmakeargs=(
+		-DCMAKE_INSTALL_DOCDIR="share/doc/${PF}"
+		-DCMAKE_CXX_FLAGS_RELEASE="${CXXFLAGS}"
```
Comment (SoapGentoo) [inline dev-embedded/ponyprog/ponyprog-3.0.0.ebuild]: don't do this - this should work by default, and if it doesn't please fix the build system
